Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -234,10 +237,24 @@ public void Delete(string id)
                     // Perform the build deployment of this changeset
                     await Build(changeSet, tracer, deployStep, repository, deploymentInfo, deploymentAnalytics, fullBuildByDefault);
 
-                    if (!(OSDetector.IsOnWindows() && !EnvironmentHelper.IsWindowsContainers()) && _settings.RestartAppContainerOnGitDeploy())
+                    if ((!OSDetector.IsOnWindows() || EnvironmentHelper.IsWindowsContainers() || deploymentInfo.ShouldRestartAfterDeployment) && _settings.RestartAppContainerOnGitDeploy())
                     {
                         logger.Log(Resources.Log_TriggeringContainerRestart);
-                        DockerContainerRestartTrigger.RequestContainerRestart(_environment, RestartTriggerReason);
+
+                        if (_settings.RecylePreviewEnabled())
+                        {
+                            logger.Log("Triggering recycle (preview mode enabled).");
+                            tracer.Trace("Triggering recycle (preview mode enabled).");
+
+                            await RestartMainSiteAsync(deploymentInfo);
+                        }
+                        else
+                        {
+                            logger.Log("Triggering recycle.");
+                            tracer.Trace("Triggering recycle.");
+
+                            DockerContainerRestartTrigger.RequestContainerRestart(_environment, RestartTriggerReason, tracer);
```

Comment:
>Docker [](start = 28, length = 6)

I think we should try not to use the word 'Docker' and just call it ContainerRestartTrigger.

Reply:
Leaving this as-is, as discussed.

Reply:
Leaving this unchanged, as discussed.

---
In reply to: [331619685](https://github.com/projectkudu/kudu/pull/3050#discussion_r331619685) [](ancestors = 331619685)